dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/file/single/SingleItemDataWriter.kt

```diff
@@ -9,16 +9,21 @@ package com.datadog.android.core.internal.persistence.file.single
 import androidx.annotation.WorkerThread
 import com.datadog.android.core.internal.persistence.DataWriter
 import com.datadog.android.core.internal.persistence.Serializer
-import com.datadog.android.core.internal.persistence.file.FileHandler
+import com.datadog.android.core.internal.persistence.file.ChunkedFileHandler
 import com.datadog.android.core.internal.persistence.file.FileOrchestrator
+import com.datadog.android.core.internal.persistence.file.FilePersistenceConfig
 import com.datadog.android.core.internal.persistence.serializeToByteArray
 import com.datadog.android.log.Logger
+import com.datadog.android.log.internal.utils.errorWithTelemetry
+import java.util.Locale
 
 internal open class SingleItemDataWriter<T : Any>(
     internal val fileOrchestrator: FileOrchestrator,
     internal val serializer: Serializer<T>,
-    internal val handler: FileHandler,
-    internal val internalLogger: Logger
+    internal val handler: ChunkedFileHandler,
```

Comment:
shouldn't be a `SingleItemFileHandler` here ?

Reply:
Ideally, but it will be a subject of RUMM-2235 anyway. Currently the only implementation of `ChunkedFileHandler` is `BatchFileHandler` which will write data in TLV format adding meta as well, which are (both meta and TLV) not really needed for the use-cases of `SingleItemDataWriter` which are `NdkUserInfoDataWriter` and `NdkNetworkInfoDataWriter` (since we just want to store plain json), but these cases will work with `BatchFileHandler` as well.